pkg/util/leaderlease/leaderlease.go

```diff
@@ -165,7 +166,10 @@ func (e *Etcd) tryHold(ttl, index uint64) error {
 			return
 		}
 		glog.V(4).Infof("Lease %s lost due to deletion at %d", e.key, watchIndex)
-		close(lost)
+		if !closedLost {
```

Comment:
That would block if no one has even sent to the channel.  You can't use a
fall through select either, because you can still race.

On Fri, May 6, 2016 at 10:57 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/util/leaderlease/leaderlease.go
> <https://github.com/openshift/origin/pull/8350#discussion_r62341075>:
>
> > @@ -165,7 +166,10 @@ func (e *Etcd) tryHold(ttl, index uint64) error {
> >  			return
> >  		}
> >  		glog.V(4).Infof("Lease %s lost due to deletion at %d", e.key, watchIndex)
> > -		close(lost)
> > +		if !closedLost {
>
> Am I the only one finding that receiving from a channel before trying to
> close it is saner than a boolean?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8350/files/bd8dd83544b2324a01d87f8c75ce057a08ed8ab0#r62341075>
>